firefox_services_tests/apps/base.py

```diff
@@ -0,0 +1,79 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from marionette_driver import expected, Wait
+from marionette_driver.errors import NoSuchElementException
+
+
+class Base(object):
+
+    def __init__(self, marionette):
+        self.marionette = marionette
+        self.marionette.timeout = 10000
+        self.set_context(self.marionette.CONTEXT_CONTENT)
+
+    def launch(self, url=None):
+        self.marionette.navigate(url)
+
+    def is_element_present(self, by, locator):
+        self.marionette.set_search_timeout(0)
+        try:
+            self.marionette.find_element(by, locator)
+            return True
+        except NoSuchElementException:
+            return False
+        finally:
+            self.marionette.set_search_timeout(
+                self.marionette.timeout or 10000)
+
+    def is_element_displayed(self, by, locator):
+        self.marionette.set_search_timeout(0)
+        try:
+            return self.marionette.find_element(by, locator).is_displayed()
+        except NoSuchElementException:
+            return False
+        finally:
+            self.marionette.set_search_timeout(
+                self.marionette.timeout or 10000)
+
+    def wait_for_element_displayed(self, by, locator):
+        return Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(expected.element_present(by, locator))))
+
+    def wait_for_element_present(self, by, locator):
+        Wait(self.marionette).until(expected.element_present(by, locator))
+
+    def wait_for_element_enabled(self, by, locator):
+        Wait(self.marionette).until(
+            expected.element_enabled(lambda m: m.find_element(by, locator)))
+
+    def wait_for_element_not_displayed(self, by, locator):
+        Wait(self.marionette).until(expected.element_not_displayed(
+            Wait(self.marionette).until(expected.element_present(by, locator))))
+
+    def wait_for_element_not_present(self, by, locator):
+        Wait(self.marionette).until(expected.element_not_present(by, locator))
+
+    def wait_for_element_not_enabled(self, by, locator):
+        Wait(self.marionette).until(
+            expected.element_not_enabled(lambda m: m.find_element(by, locator)))
+
+    def set_context(self, context):
+        if context != 'chrome' and context != 'content':
```

Comment:
The strings here would be better served as constants e.g. `self.CHROME = 'chrome'` and `self.CONTENT = 'content'` as people can then use the constants and not the mysterious strings